bfs.go

```diff
@@ -56,3 +59,36 @@ type Iterator interface {
 type supportsCopying interface {
 	Copy(context.Context, string, string) error
 }
+
+// --------------------------------------------------------------------
+
+// Resolve opens a bucket from a URL string.
+func Resolve(ctx context.Context, u *url.URL) (Bucket, error) {
+	registryLock.Lock()
```

Comment:
Minor: that's fine, but I'd not overcomplicate and just tell in README (or better in godoc comment) that's it's not thread safe and should be called only in `func init()`. Or you expect these to be concurrent as well?

Reply:
Also there's a sync.Map, as I remember - probs, worth being used here to get rid of mutex.

Reply:
I don't see a reason to ditch the mutex, it's simple and easy. also package loading in Go is not deterministic, if someone does:

```
import (
 _ "github.com/bfs/bfss3"
 _ "github.com/bfs/bfsgs"
)
```

Can anyone be absolutely sure that they are not loaded in parallel? Better: keep mutex!